### ISSUE TITLE
How to add my dataset to MMF for using existing multimodal models?

### ISSUE THREAD

**Issue by KushRJ** [facebookresearch/mmf#475]
## ❓ Questions and Help
Forgive me but I am new to this package. I went through the official tutorial on how to add a dataset, but it is quite complex with multiple stages, as shown here: https://mmf.sh/docs/tutorials/dataset/ 
The webpage also says that '[Outdated] A new version of this will be uploaded soon', so is there any simple way to add my dataset comprising of text and images?

A fast response is highly appreciated. TIA!

**Comment by apsdehal** [comment]
Hi,
More details on what the dataset is about would be helpful 

For new tutorial have a look at this colab notebook where we add OKVQA dataset from scratch: https://colab.research.google.com/github/facebookresearch/mmf/blob/notebooks/notebooks/kdd_tutorial.ipynb

**Comment by KushRJ** [comment]
The dataset is Twitter data with tweets including image and text together.

**Comment by apsdehal** [comment]
Please have a look at the colab. You can follow the instructions in it to implement your own dataset.

**Comment by gireek** [comment]
Hi @apsdehal Is there a colab for loading a dataset on similar lines with the hateful memes dataset where there is an image and a text and you just have to classify it into multiple classes . Thanks

**Comment by apsdehal** [comment]
No, we don't have such a colab at the moment. But if dataset is similar to hateful memes, you can just use its code? Also, OKVQA example is already multiple classes and probably be adjusted for your use case?

**Comment by gireek** [comment]
@apsdehal In my training case the last line in the logs is : 

`/root/.cache/torch/mmf/wiki.en.bin 2020-08-20T17:28:04 | INFO | mmf.train : Finished loading fasttext model`

but the ./save/models folder is empty. No models have been saved. 
I have also raised an issue here: https://github.com/facebookresearch/mmf/issues/501
Kindly check

**Comment by gireek** [comment]
The kind of rows in train.jsonl in hateful memes dataset are like :
`{"id":13894,"img":"img\/13894.png","label":0,"text":"putting bows on your pet"}
`

but in my data they are like:
`{"id": 724, "img": "./img/724.jpg", "label": 2, "text": "KALE GREEN TEA "}
`

the difference being in the key "img" in the 2 rows. Can that be a problem because of paths?


Even the folder structure i have is same as the hateful memes dataset being:
./data
       |------- img
       |------- train.jsonl
       |------- test.jsonl
       |------- dev.jsonl

Then, I zipped the data folder with a password becuase it seems MMF commands need a password secured zip file dataset by:

`!zip -r --password pass123 ./training.zip ./data/`

and then opened it as an MMF dataset:

`!mmf_convert_hm --zip_file=./training.zip --password=pass123 --bypass_checksum=1`

**Comment by arjondas** [comment]
Try: 
!pip install --pre --upgrade mmf torch==1.5.0+cu101 torchvision==0.6.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html

Although I'm not sure if it's gonna break mmf

**Comment by sharduls007** [comment]
HI @apsdehal,
I am trying to run colab notebook in this thread but getting below error while running run(opts)
![image](https://user-images.githubusercontent.com/37594733/100223098-ef8f6100-2f12-11eb-975e-d6fb3a2d99a7.png)

**Comment by Mayor2305** [comment]
> Hi, More details on what the dataset is about would be helpful
> 
> For new tutorial have a look at this colab notebook where we add OKVQA dataset from scratch: https://colab.research.google.com/github/facebookresearch/mmf/blob/notebooks/notebooks/kdd_tutorial.ipynb

Hello, 

I was trying this notebook but after running the last cell (code below) of building the dataset section:

from mmf.utils.build import build_dataset
from mmf.utils.configuration import Configuration

# Init configuration to register resolvers
Configuration()
dataset = build_dataset("okvqa_colab", dataset_type="val")
import matplotlib.pyplot as plt
plt.rcParams["figure.figsize"] = (20, 20)
dataset.visualize(num_samples=8, size=(512, 512), nrow=4)


I got an error: 

FileNotFoundError: [Errno 2] No such file or directory: '/root/.cache/torch/mmf/data/datasets/okvqa_colab/defaults/annotations/annotations/imdb_val1.npy'
 
Please help me regarding the same...

ps: i am using google colab as the environment
